src/averageAges.js

```diff
@@ -55,7 +70,18 @@ function calculateWomenAverageAge(people, withChildren) {
  * @return {number}
  */
 function calculateAverageAgeDiff(people, onlyWithSon) {
-  // write code here
+  const kids = people.filter(kid => (
+    onlyWithSon
+      ? people.find(mother => mother.name === kid.mother)
```

Comment:
You are duplicating this expression `people.find(mother => mother.name === kid.mother)`. Maybe, let's create the variable for it?